checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,990 @@
+    public boolean isCollections(TypeMirror tm) {
+        return types.isSameType(tm, CollectionsTypeMirror);
+    }
+
+    /** @return true if {@code tm} is AbstractList or a subtype of AbstractList */
+    public boolean isAbstractList(TypeMirror tm) {
+        return types.isSubtype(types.erasure(tm), types.erasure(AbstractListTypeMirror));
+    }
+
+    /**
+     * @return true if {@code tm} is AbstractSequentialList or a subtype of AbstractSequentialList
+     */
+    public boolean isAbstractSequentialList(TypeMirror tm) {
+        return types.isSubtype(types.erasure(tm), types.erasure(AbstractSequentialListTypeMirror));
+    }
+
+    /** @return true if {@code tm} is ArrayList or a subtype of ArrayList */
+    public boolean isArrayList(TypeMirror tm) {
+        return types.isSubtype(types.erasure(tm), types.erasure(ArrayListTypeMirror));
+    }
+
+    /** @return true if {@code tm} is LinkedList or a subtype of LinkedList */
+    public boolean isLinkedList(TypeMirror tm) {
+        return types.isSubtype(types.erasure(tm), types.erasure(LinkedListTypeMirror));
+    }
+
+    /** @return true if {@code tm} is NavigableSet or a subtype of NavigableSet */
+    public boolean isNavigableSet(TypeMirror tm) {
+        return types.isSubtype(types.erasure(tm), types.erasure(NavigableSetTypeMirror));
+    }
+
+    /** @return true if {@code tm} is SortedSet or a subtype of SortedSet */
+    public boolean isSortedSet(TypeMirror tm) {
+        return types.isSubtype(types.erasure(tm), types.erasure(SortedSetTypeMirror));
+    }
+
+    /** @return true if {@code tm} is TreeSet or a subtype of TreeSet */
+    public boolean isTreeSet(TypeMirror tm) {
+        return types.isSubtype(types.erasure(tm), types.erasure(TreeSetTypeMirror));
+    }
+
+    /** @return true if {@code tm} is Enumeration or a subtype of Enumeration */
+    public boolean isEnumeration(TypeMirror tm) {
+        return types.isSubtype(types.erasure(tm), types.erasure(EnumerationTypeMirror));
+    }
+
+    /**
+     * Returns true if {@code receiverUnderlyingType} is Iterator and if the return type of {@code
+     * invokedMethodElement} is a type variable.
+     */
+    private boolean isIteratorNext(
+            TypeElement receiverUnderlyingType, ExecutableElement invokedMethodElement) {
+        if (isIterator(receiverUnderlyingType.asType())) {
+            if (invokedMethodElement.getSimpleName().contentEquals("next")
+                    && invokedMethodElement.getReturnType().getKind() == TypeKind.TYPEVAR
+                    && invokedMethodElement.getParameters().isEmpty()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if {@code receiverUnderlyingType} is AbstractList, AbstractSequentialList, or
+     * List and if the return type of {@code invokedMethodElement} is a type variable.
+     */
+    private boolean isAbstractListWithTypeVarReturn(
```

Comment:
Much of the code in this method can be replaces with `org.checkerframework.javacutil.ElementUtils#isMethod`.